taichi/backends/vulkan/vulkan_device.cpp

```diff
@@ -190,22 +190,37 @@ void VulkanResourceBinder::framebuffer_depth_stencil(DeviceAllocation image) {
 void VulkanResourceBinder::write_to_set(uint32_t index,
                                         VulkanDevice &device,
                                         VkDescriptorSet set) {
-  std::vector<VkDescriptorBufferInfo> buffer_info;
+  std::vector<VkDescriptorBufferInfo> buffer_infos;
   std::vector<VkWriteDescriptorSet> desc_writes;
```

Comment:
Just to be safe, we can `reserve()` in the beginning.

Reply:
Since I won't know the size until I went through all the descriptors, I don't think we can use reserve here? Using a second pass should work.